lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -911,30 +942,466 @@ class ConvertAtenNativeLayerNormOp
+      expandShape[i] = -1;
+    }
+    auto expandShapeType = RankedTensorType::get(expandShape, elemTy);
+    SmallVector<ReassociationIndices> reassociation(meanAndVarShapeRank);
+    for (auto i : llvm::seq<int64_t>(0, meanAndVarShapeRank)) {
+      if (i != meanAndVarShapeRank - 1) {
+        reassociation[i].push_back(i);
+      } else {
+        reassociation[i].push_back(i);
+        for (auto j : llvm::seq<int64_t>(0, normalizedShapeRank))
+          reassociation[i].push_back(i + j + 1);
+      }
+    }
+    Value meanResult = rewriter.create<tensor::ExpandShapeOp>(
+        loc, expandShapeType, mean, reassociation);
+    Value rSTDResult = rewriter.create<tensor::ExpandShapeOp>(
+        loc, expandShapeType, rSTD, reassociation);
     Type layerNormResultType = getTypeConverter()->convertType(op.getType(0));
     Type meanResultType = getTypeConverter()->convertType(op.getType(1));
-    Type varResultType = getTypeConverter()->convertType(op.getType(2));
+    Type rSTDResultType = getTypeConverter()->convertType(op.getType(2));
     Value layerNorm_ =
         rewriter.create<tensor::CastOp>(loc, layerNormResultType, layerNorm);
-    Value mean_ = rewriter.create<tensor::CastOp>(loc, meanResultType, mean);
-    Value var_ = rewriter.create<tensor::CastOp>(loc, varResultType, var);
+    Value mean_ =
+        rewriter.create<tensor::CastOp>(loc, meanResultType, meanResult);
+    Value var_ =
+        rewriter.create<tensor::CastOp>(loc, rSTDResultType, rSTDResult);
     rewriter.replaceOp(op, {layerNorm_, mean_, var_});
     return success();
   }
 };
 } // namespace
 
+namespace {
+
+// Converts the `aten.native_layer_norm_backward` op to linalg. There are three
+// components to be calculated here:
+//  1) Gradient of the weights(dWeight).
+//  2) Gradiend of the bias(dBias).
+//  3) Gradient of the input(dInput).
+class ConvertAtenNativeLayerNormBackwardOp
+    : public OpConversionPattern<AtenNativeLayerNormBackwardOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult
+  matchAndRewrite(AtenNativeLayerNormBackwardOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    MLIRContext *context = op->getContext();
+    Location loc = op->getLoc();
+    Value gradOut = adaptor.grad_out();
+    Value input = adaptor.input();
+    Value normalizedShape = adaptor.normalized_shape();
+    Value mean = adaptor.mean();
+    Value rSTD = adaptor.rstd();
+    Value weight = adaptor.weight();
+    Value bias = adaptor.bias();
+
+    if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
+      return failure();
+
+    // TODO: Handle the None cases for the optional parameters:
+    // weight, bias.
+    if (failed(checkNotNone(rewriter, op, gradOut)) ||
+        failed(checkNotNone(rewriter, op, input)) ||
+        failed(checkNotNone(rewriter, op, mean)) ||
+        failed(checkNotNone(rewriter, op, rSTD)) ||
+        failed(checkNotNone(rewriter, op, weight)) ||
```

Comment:
Only `weight` and `bias` are optional so only need to check those two to be not none.